pkg/gzip/gzip.go

```diff
@@ -0,0 +1,159 @@
+package gzip
```

Comment:
i made this a new package because it creates an import cycle if it's in `util` (`util` imports `user` and vice-versa)